client/lib/signup/flow-controller.js

```diff
@@ -205,11 +246,12 @@ assign( SignupFlowController.prototype, {
 			SignupActions.processSignupStep( step );
 
 			const apiFunction = steps[ step.stepName ].apiRequestFunction.bind( this );
-
-			apiFunction( ( errors, providedDependencies ) => {
+			const apiFunctionDoneCallback = ( errors, providedDependencies ) => {
 				this._processingSteps[ step.stepName ] = false;
 				SignupActions.processedSignupStep( step, errors, providedDependencies );
-			}, dependenciesFound, step );
+			};
+
+			defer( apiFunction, apiFunctionDoneCallback, dependenciesFound, step );
```

Comment:
Yes, if we want the apiFunction to be able to dispatch action, since we are in a call graph that started in a dispatch handler.